rcl_lifecycle/src/transition_map.c

```diff
@@ -117,7 +117,7 @@ rcl_lifecycle_register_transition(
   if (!state) {
     char * error_msg = rcutils_format_string(rcl_get_default_allocator(),
         "state %u is not registered\n", transition.start->id);
-    RCL_SET_ERROR_MSG(error_msg, rcl_get_default_allocator());
+    rcl_set_formatted_error(error_msg, rcl_get_default_allocator());
```

Comment:
These two are definitely memory leaks, but can be cleaned up after setting the error state without a new function, i.e. `rcl_set_formatted_error`.

Reply:
@Karsten1987 fyi

Reply:
the `rcl_set_formatted_error` is just a macro to `rcutils_set_formatted_error`, so this more or less depends the conclusion for `rcutils_set_formatted_error` in PR https://github.com/ros2/rcutils/pull/63 to be unified. thanks